### PR TITLE
Fixing lint issues, update package name and go_package, update make api

### PR DESCRIPTION
**Pull request by akoserwal**
* Fix lint issue based on `buf lint`
* Update the proto package for `kessel.inventory.v1beta1;`
* Update option go_package = "github.com/project-kessel/inventory-api/api/kessel/inventory/v1beta1";
* Update `make api` to use docker container

### CODE REVIEW

File: api/kessel/inventory/v1beta1/k8s_cluster_detail.proto [hunk 1]
```diff
@@ -14,59 +14,59 @@ message K8sClusterDetail {
...
-    Cloud_platformEnum_OPENSTACK_UPI = 12;
-    Cloud_platformEnum_GCP_IPI = 13;
-    Cloud_platformEnum_GCP_UPI = 14;
-    Cloud_platformEnum_NUTANIX_IPI = 15;
-    Cloud_platformEnum_NUTANIX_UPI = 16;
-    Cloud_platformEnum_VSPHERE_IPI = 17;
-    Cloud_platformEnum_VSPHERE_UPI = 18;
-    Cloud_platformEnum_OVIRT_IPI = 19;
+  enum CloudPlatformEnum {
+    CLOUD_PLATFORM_ENUM_UNKNOWN_IPI_UNSPECIFIED = 0;
```
Comment (jewzaam) [inline api/kessel/inventory/v1beta1/k8s_cluster_detail.proto:47]: If unknown why indicate specifically IPI?

```suggestion
    CLOUD_PLATFORM_ENUM_UNKNOWN_UNSPECIFIED = 0;
```

File: api/kessel/inventory/v1beta1/policy_detail.proto [hunk 2]
```diff
@@ -13,11 +13,11 @@ message PolicyDetail {
...
-    SeverityEnum_LOW = 0;
-    SeverityEnum_MEDIUM = 1;
-    SeverityEnum_HIGH = 2;
-    SeverityEnum_CRITICAL = 3;
-    SeverityEnum_OTHER = 4;
+    SEVERITY_ENUM_LOW_UNSPECIFIED = 0;
+    SEVERITY_ENUM_MEDIUM = 1;
+    SEVERITY_ENUM_HIGH = 2;
+    SEVERITY_ENUM_CRITICAL = 3;
+    SEVERITY_ENUM_OTHER = 4;
```
Comment (jewzaam) [inline api/kessel/inventory/v1beta1/policy_detail.proto:20]:
```suggestion
    SEVERITY_ENUM_UNSPECIFIED = 0;
    SEVERITY_ENUM_LOW = 1;
    SEVERITY_ENUM_MEDIUM = 2;
    SEVERITY_ENUM_HIGH = 3;
    SEVERITY_ENUM_CRITICAL = 4;
    SEVERITY_ENUM_OTHER = 5;
```

I wouldn't assume "low" and "unspecified" are synonymous.  Use separate values.

File: api/kessel/inventory/v1beta1/reporter_data.proto [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
-    Reporter_typeEnum_HBI = 1;
-    Reporter_typeEnum_OCM = 2;
-    Reporter_typeEnum_OTHER = 3;
-    Reporter_typeEnum_UNKNOWN = 4;
+  enum ReporterTypeEnum {
+    REPORTER_TYPE_ENUM_ACM_UNSPECIFIED = 0;
+    REPORTER_TYPE_ENUM_HBI = 1;
+    REPORTER_TYPE_ENUM_OCM = 2;
+    REPORTER_TYPE_ENUM_OTHER = 3;
+    REPORTER_TYPE_ENUM_UNKNOWN_UNSPECIFIED = 4;
```
Comment (jewzaam) [inline api/kessel/inventory/v1beta1/reporter_data.proto:19]:
```suggestion
    REPORTER_TYPE_ENUM_UNKNOWN = 0;
    REPORTER_TYPE_ENUM_ACM = 1;
    REPORTER_TYPE_ENUM_HBI = 2;
    REPORTER_TYPE_ENUM_OCM = 3;
    REPORTER_TYPE_ENUM_OTHER = 4;
```

Don't overload ACM and unknown.  I've updated to "UNKNOWN" based on previous state.  This is not consistent with other enums which use "UNKNOWN_UNSPECIFIED".  I'd prefer we pick one word/phrase and stick to it for all enums, I don't have a preference on what specifically.

**Review by jewzaam — CHANGES_REQUESTED** [review]

File: api/kessel/inventory/v1beta1/reporter_data.proto (hunk 3 above)
Comment (akoserwal) [reply api/kessel/inventory/v1beta1/reporter_data.proto:19]: Buf lint suggest for `UNKNOWN`  to append a suffix for UNSPECIFIED 

eg:` kessel/inventory/v1beta1/reporter_data.proto:15:5:Enum zero value name "Reporter_typeEnum_ACM" should be suffixed with "_UNSPECIFIED".`

we can `REPORTER_TYPE_ENUM_UNKNOWN` to `REPORTER_TYPE_ENUM_UNSPECIFIED`

Comment (jewzaam) [reply api/kessel/inventory/v1beta1/reporter_data.proto]: Sure, using `UNSPECIFIED` works.  So a few replaces?  `/UNKNOWN/UNSPECIFIED/` and `/UNKNOWN_UNSPECIFIED/UNSPECIFIED/`?

**Review by jewzaam — APPROVED** [review]

**Comment by alechenninger** [comment]
Gonna go ahead and merge with latest updates. Please open issues / slack threads / PRs for follow ups.

**Pull request merged**